File: server-scala-play-server/app/model/RetailerLocation.scala
package model

import play.api.libs.json._
import java.time.OffsetDateTime

/**
  * Represents the Swagger definition for RetailerLocation.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class RetailerLocation(
  id: Option[Long],
  active: Option[Boolean],
  valid: Option[Boolean],
  latitude: Option[Double],
  longitude: Option[Double],
  locationDescription: Option[String],
  categories: Option[List[Category]],
  filters: Option[List[Filter]],
  offerLocations: Option[List[OfferLocation]],
  logo: Option[Asset],
  picture1: Option[Asset],
  picture2: Option[Asset],
  qrCode: Option[Asset],
  name: Option[String],
  locationId: Option[String],
  code: Option[String],
  retailer: Option[Retailer],
  assignments: Option[List[Assignment]],
  currentAssignment: Option[Assignment],
  profile: Option[RetailerProfile],
  regions: Option[List[Region]],
  offerPrintKeyPrefix: Option[String],
  administrators: Option[List[Account]],
  sqootId: Option[Long],
  yipitId: Option[Long],
  locationToken: Option[String],
  building: Option[Building],
  notes: Option[List[Note]],
  noteCount: Option[Long],
  searchIndexUpdated: Option[OffsetDateTime],
  inSearchIndex: Option[Boolean],
  favoriteCount: Option[Long],
  hasRatings: Option[Boolean],
  googlePlaceId: Option[String],
  yelpId: Option[String],
  display: Option[String],
  appKey: Option[String],
  categoryTree: Option[String],
  filterTree: Option[String],
  addressDisplay: Option[String],
  mapQuery: Option[String],
  sortName: Option[String],
  fullDisplay: Option[String],
  hasOffers: Option[Boolean],
  billableId: Option[Long],
  subType: Option[String],
  contentName: Option[String],
  owner: Option[Account],
  secondaryType: Option[String],
  locationAddress: Option[String],
  validOfferLocationIds: Option[List[Long]]
)

object RetailerLocation {
  implicit lazy val retailerLocationJsonFormat: Format[RetailerLocation] = Json.format[RetailerLocation]
}

File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/RetailerLocation.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class RetailerLocation 
{
    public long Id { get; set; }
    public bool Active { get; set; }
    public bool Valid { get; set; }
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public string LocationDescription { get; set; }
    public List<Category> Categories { get; set; }
    public List<Filter> Filters { get; set; }
    public List<OfferLocation> OfferLocations { get; set; }
    public Asset Logo { get; set; }
    public Asset Picture1 { get; set; }
    public Asset Picture2 { get; set; }
    public Asset QrCode { get; set; }
    public string Name { get; set; }
    public string LocationId { get; set; }
    public string Code { get; set; }
    public Retailer Retailer { get; set; }
    public List<Assignment> Assignments { get; set; }
    public Assignment CurrentAssignment { get; set; }
    public RetailerProfile Profile { get; set; }
    public List<Region> Regions { get; set; }
    public string OfferPrintKeyPrefix { get; set; }
    public List<Account> Administrators { get; set; }
    public long SqootId { get; set; }
    public long YipitId { get; set; }
    public string LocationToken { get; set; }
    public Building Building { get; set; }
    public List<Note> Notes { get; set; }
    public long NoteCount { get; set; }
    public DateTime SearchIndexUpdated { get; set; }
    public bool InSearchIndex { get; set; }
    public long FavoriteCount { get; set; }
    public bool HasRatings { get; set; }
    public string GooglePlaceId { get; set; }
    public string YelpId { get; set; }
    public string Display { get; set; }
    public string AppKey { get; set; }
    public string CategoryTree { get; set; }
    public string FilterTree { get; set; }
    public string AddressDisplay { get; set; }
    public string MapQuery { get; set; }
    public string SortName { get; set; }
    public string FullDisplay { get; set; }
    public bool HasOffers { get; set; }
    public long BillableId { get; set; }
    public string SubType { get; set; }
    public string ContentName { get; set; }
    public Account Owner { get; set; }
    public string SecondaryType { get; set; }
    public string LocationAddress { get; set; }
    public List<long> ValidOfferLocationIds { get; set; }
}


